### PR TITLE
Add revert message when burnFrom exceeds allowance

### PR DESCRIPTION
**Pull request by vanruch**

### CODE REVIEW

File: contracts/truefi2/DeficiencyToken.sol [hunk 1]
```diff
@@ -31,7 +31,11 @@ contract DeficiencyToken is IDeficiencyToken, ERC20 {
     }
 
     function burnFrom(address account, uint256 amount) external override {
-        _approve(account, _msgSender(), allowance(account, _msgSender()).sub(amount));
+        _approve(
+            account,
+            _msgSender(),
+            allowance(account, _msgSender()).sub(amount, "DeficiencyToken: burn amount exceeds allowance")
```
Comment (barrutko) [inline contracts/truefi2/DeficiencyToken.sol:37]:
```suggestion
            allowance(account, _msgSender()).sub(amount, "DeficiencyToken: Burn amount exceeds allowance")
```

File: test/truefi2/DeficiencyToken.test.ts [hunk 2]
```diff
@@ -51,13 +51,13 @@ describe('DeficiencyToken', () => {
     it('reverts excessive burn from allowed account', async () => {
       await deficiency.mint(owner.address, parseEth(3))
       await deficiency.connect(owner).approve(friend.address, parseEth(2))
-      await expect(deficiency.connect(friend).burnFrom(owner.address, parseEth(2).add(1))).to.be.revertedWith('revert SafeMath: subtraction overflow')
+      await expect(deficiency.connect(friend).burnFrom(owner.address, parseEth(2).add(1))).to.be.revertedWith('DeficiencyToken: burn amount exceeds allowance')
```
Comment (barrutko) [inline test/truefi2/DeficiencyToken.test.ts:54]:
```suggestion
      await expect(deficiency.connect(friend).burnFrom(owner.address, parseEth(2).add(1))).to.be.revertedWith('DeficiencyToken: Burn amount exceeds allowance')
```

File: test/truefi2/DeficiencyToken.test.ts [hunk 3]
```diff
@@ -51,13 +51,13 @@ describe('DeficiencyToken', () => {
...
     })
 
     it('reverts burn from unrelated account', async () => {
       await deficiency.mint(owner.address, parseEth(3))
       await deficiency.connect(owner).approve(friend.address, parseEth(2))
-      await expect(deficiency.connect(stranger).burnFrom(owner.address, 1)).to.be.revertedWith('revert SafeMath: subtraction overflow')
+      await expect(deficiency.connect(stranger).burnFrom(owner.address, 1)).to.be.revertedWith('DeficiencyToken: burn amount exceeds allowance')
```
Comment (barrutko) [inline test/truefi2/DeficiencyToken.test.ts:60]:
```suggestion
      await expect(deficiency.connect(stranger).burnFrom(owner.address, 1)).to.be.revertedWith('DeficiencyToken: Burn amount exceeds allowance')
```

**Comment by yuchenlintt** [comment]
Thanks for the fix! Much better error message.

**Review by yuchenlintt — APPROVED** [review]

**Pull request merged**